### PR TITLE
fix #1719 Add Google Compute Engine support

### PR DESCRIPTION
**Pull request by rvrignaud**
This is a first step in GCE support.
- This requires a very some patches only on fog's master branch.
- This does not support yet instances with boot disk 

Any thoughts ?

### CODE REVIEW

File: app/models/compute_resources/foreman/model/gce.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      client.flavors
+    end
+
+    def vms
+      client.servers
+    end
+
+    def create_vm args = {}
+      #Dot are not allowed in names
+      args['name'] = args['name'].split('.').first
```
Comment (ohadlevy) [inline app/models/compute_resources/foreman/model/gce.rb:33]: you can use the parametize method, we used to do the same in ovirt

File: bundler.d/gce.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+group :gce do
+  gem "google-api-client"
+  gem 'sshkey'
```
Comment (lzap) [inline bundler.d/gce.rb:3]: Just a heads up - new dependency: https://rubygems.org/gems/sshkey

(We might want to reuse this dependency for more ssh generation)

File: app/models/compute_resources/foreman/model/gce.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
+module Foreman::Model
+  class GCE < ComputeResource
+    has_one :key_pair, :foreign_key => :compute_resource_id
+    after_create :setup_key_pair
+    after_destroy :destroy_key_pair
+
+    def capabilities
+	[:image]
```
Comment (skottler) [inline app/models/compute_resources/foreman/model/gce.rb]: The indentation is off here.

**Comment by skottler** [comment]
`app/models/compute_resources/foreman/model/gce.rb` has a lot of indentation issues. I'll test this today with my GCE account and report back.

**Comment by rvrignaud** [comment]
@skottler : thank you for your test !
- To get the right version of fog, I suggest you to use my branch 
  (https://github.com/rvrignaud/fog :branch => 'foreman')
- Be sure that Foreman is able to reach :private_ip_address or change provided_attributes
- Be sure to test with an instance that doesn't need a boot disk (I'm working on that suppor) ie n1-standard-1
- Do not use europe-west1-a zone which is down for now

**Comment by ohadlevy** [comment]
merged as d1fa5fa thanks @rvrignaud, will be part of 1.3.

**Pull request closed without merging**